drivers/led_strip/ws2812.c

```diff
@@ -47,9 +47,16 @@
  */
 #define RESET_NFRAMES ((size_t)ceiling_fraction(3 * SPI_FREQ, 4000000) + 1)
 
+#if CONFIG_WS2812_HAS_WHITE_CHANNEL
+#define PX_BUF_PER_PX 32
+#else
+#define PX_BUF_PER_PX 24
+#endif
+
 struct ws2812_data {
 	struct device *spi;
 	struct spi_config config;
+	u8_t px_buf[PX_BUF_PER_PX * CONFIG_WS2812_STRIP_MAX_PIXELS];
```

Comment:
how come this was hardcoded to 32 and now it's 32 (or 24) times MAX_PIXELS?

Reply:
It used to be a single pixel's worth of data, in a representation suitable to feed to the SPI driver. Now it's an entire pixel strip's worth of SPI data. A pixel takes up either 32B or 24B depending on a configuration option. Back when the difference was only 8B, I just left it at the max value. Now that it's 8B * NUM_PIXELS, I'm not.

Reply:
(I know this is gross amounts of memory; happy to get into the details on how this could be fixed with GPIO driver API extensions if you ever want to)